### PR TITLE
chore: tighten tekton controller rbac permissions

### PR DESCRIPTION
**Pull request by eddycharly**
This PR removes unnecessary rbac permissions to the tekton controller.

### CODE REVIEW

**Comment by eddycharly** [comment]
Did i cut too much rights ?

File: charts/lighthouse/templates/tekton-controller-role.yaml [hunk 1]
```diff
@@ -5,50 +5,33 @@ metadata:
...
   verbs:
   - get
-  - list
-  - watch
 - apiGroups:
   - tekton.dev
   resources:
-  - pipelineresources
-  - tasks
-  - pipelines
```
Comment (abayer) [inline charts/lighthouse/templates/tekton-controller-role.yaml:22]: Yup, we still need `tasks` and `pipelines` for sure.

**Comment by abayer** [comment]
ah, more specifically:

```
E0827 16:17:53.462005       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.6/tools/cache/reflector.go:105: Failed to list *v1beta1.Pipeline: pipelines.tekton.dev is forbidden: User "system:serviceaccount:lh-test:lighthouse-tekton-controller" cannot list resource "pipelines" in API group "tekton.dev" in the namespace "lh-test"
```

File: charts/lighthouse/templates/tekton-controller-role.yaml [hunk 2]
```diff
@@ -5,50 +5,33 @@ metadata:
...
-  - ""
+  - tekton.dev
   resources:
-  - namespaces
-  - configmaps
-  - secrets
+  - pipelines
   verbs:
   - get
-  - list
```
Comment (abayer) [inline charts/lighthouse/templates/tekton-controller-role.yaml:15]: Yup, need this back.

Comment (eddycharly) [reply charts/lighthouse/templates/tekton-controller-role.yaml:15]: Hmmm yeah... controller-runtime cache again, I need a non caching client instead.

Comment (eddycharly) [reply charts/lighthouse/templates/tekton-controller-role.yaml:15]: Should be fixed 🤞

File: charts/lighthouse/templates/tekton-controller-role.yaml [hunk 3]
```diff
@@ -5,50 +5,33 @@ metadata:
...
-  - delete
 - apiGroups:
   - lighthouse.jenkins.io
   resources:
   - lighthousejobs
   verbs:
-  - create
-  - delete
-  - list
-  - update
```
Comment (abayer) [inline charts/lighthouse/templates/tekton-controller-role.yaml:40]: Looks like we still need this -

```
{"component":"lighthouse-tekton-controller","controller":"tekton-controller","file":"/workspace/source/pkg/engines/tekton/controller.go:131","func":"github.com/jenkins-x/lighthouse/pkg/engines/tekton.(*LighthouseJobReconciler).Reconcile","level":"error","msg":"failed to update Project status: lighthousejobs.lighthouse.jenkins.io \"700713fa-e887-11ea-bba5-f6ebb3fd55d9\" is forbidden: User \"system:serviceaccount:lh-test:lighthouse-tekton-controller\" cannot update resource \"lighthousejobs\" in API group \"lighthouse.jenkins.io\" in the namespace \"lh-test\"","time":"2020-08-27T17:05:35Z"}
```

Specifically, it's https://github.com/jenkins-x/lighthouse/blob/8fe7a143c8b56ad75c6a7acbd6dcdf2944d17b58/pkg/engines/tekton/controller.go#L127-L131 - we're probably being too aggressive in that update call (no need to make the update if we haven't actually changed the label, e.g.), but we do need update rights.

**Comment by abayer** [comment]
(the `github` failure is just transient noise and can be ignored)

**Pull request merged**